src/main/java/com/questhelper/helpers/quests/atfirstlight/AtFirstLight.java

```diff
@@ -0,0 +1,356 @@
+import com.questhelper.rewards.UnlockReward;
+import com.questhelper.steps.ConditionalStep;
+import com.questhelper.steps.DetailedQuestStep;
+import com.questhelper.steps.ItemStep;
+import com.questhelper.steps.NpcStep;
+import com.questhelper.steps.ObjectStep;
+import com.questhelper.steps.QuestStep;
+import net.runelite.api.ItemID;
+import net.runelite.api.NpcID;
+import net.runelite.api.NullObjectID;
+import net.runelite.api.ObjectID;
+import net.runelite.api.QuestState;
+import net.runelite.api.Skill;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.runelite.api.coords.WorldPoint;
+
+@QuestDescriptor(
+	quest = QuestHelperQuest.AT_FIRST_LIGHT
+)
+public class AtFirstLight extends BasicQuestHelper
+{
+	//Items Required
+	ItemRequirement needle, boxTrap, hammer, jerboaTail, jerboaTailOrBoxTrap, jerboaTail2OrBoxTrap;
+
+	// Items Recommended
+	ItemRequirement staminaPotion;
+
+	// Quest Items
+	ItemRequirement toyMouse, toyMouseWound, smoothLeaf, stickyLeaf, makeshiftPoultice, furSample, trimmedFur, foxsReport;
+
+	QuestStep talkToApatura, goDownTree, talkToVerity, talkToWolf, windUpToy, useToyOnKiko,
+		checkBed, returnToWolf, goUpTree, talkToFox, takeLeaf, takeSecondLeaf, catchJerboa, useTailOnLeaves,
+		returnToFox, talkToFoxAfterPoultice, talkToAtza, talkToAtzaAfterHandingFur, makeEquipmentPile, talkToAtzaForTrim,
+		returnToFoxAfterTrim, getReportFromFox, goDownTreeEnd, talkToVerityEnd, useJerboaTailOnBed, goUpTreeToFinishQuest,
+		talkToApaturaToFinishQuest;
+
+	QuestStep buyBoxTrap, takeNeedle, takeHammer;
+
+	//Zones
+	Zone guild;
+
+	Requirement inGuild, gotMouse, usedMouse, checkedBed, equipmentUsable, repairedEquipment, hadReport, handedInReport;
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		Map<Integer, QuestStep> steps = new HashMap<>();
+		setupZones();
+		setupRequirements();
+		setupConditions();
+		setupSteps();
+
+		// 9841 0->1?? Birb?
+		steps.put(0, talkToApatura);
+
+		ConditionalStep enterTree =  new ConditionalStep(this, goDownTree);
+
+		ConditionalStep goTalkToVerity = enterTree.copy();
+		goTalkToVerity.addStep(inGuild, talkToVerity);
+		steps.put(1, goTalkToVerity);
+
+		ConditionalStep goTalkToWolf = enterTree.copy();
+		goTalkToWolf.addStep(inGuild, talkToWolf);
+		steps.put(2, goTalkToWolf);
+
+		ConditionalStep goDistractCat = enterTree.copy();
+		goDistractCat.addStep(and(inGuild, checkedBed), returnToWolf);
+		goDistractCat.addStep(and(inGuild, usedMouse), checkBed);
+		goDistractCat.addStep(and(inGuild, toyMouseWound), useToyOnKiko);
+		goDistractCat.addStep(and(inGuild, gotMouse), windUpToy);
+		goDistractCat.addStep(inGuild, talkToWolf);
+		steps.put(3, goDistractCat);
+
+		ConditionalStep goFindFox = new ConditionalStep(this, buyBoxTrap);
+		goFindFox.addStep(boxTrap, talkToFox);
+		goFindFox.addStep(inGuild, goUpTree);
```

Comment:
Reverse the order of these to ensure the user is prompted to walk upstairs if they already have a box trap
![not-prompted-to-walk-upstairs](https://github.com/Zoinkwiz/quest-helper/assets/962989/660a72fa-5dce-431e-840c-a009285a7921)

```suggestion
		goFindFox.addStep(inGuild, goUpTree);
		goFindFox.addStep(boxTrap, talkToFox);
```